common/src/jni/main/cpp/conscrypt/native_crypto.cc

```diff
@@ -4861,6 +4861,26 @@ static jobjectArray NativeCrypto_get_X509_GENERAL_NAME_stack(JNIEnv* env, jclass
     return joa.release();
 }
 
+/*
+ * Checks an ASN1_TIME for validity, converts to epoch time in milliseconds.
+ */
+static jlong ASN1_TIME_check_and_convert_to_posix(JNIEnv* env, const ASN1_TIME* time) {
+    if (!ASN1_TIME_check(time)) {
```

Comment:
`ASN1_TIME_check` is, under the hood, implemented with the same[*] function as `ASN1_TIME_to_posix`. So you can just remove this and stick to `ASN1_TIME_to_posix`.

[*] Mostly. `ASN1_TIME_check` tolerates non-UTC time zones in the (confusingly-named) UTCTime type, but not the GeneralizedTime type due to some legacy mess. `ASN1_TIME_to_posix` is strict.

Reply:
Ah right, I misread the code and though it was needed for the ASN.1 type checking.  Thanks!